ladybug/src/main/java/org/frankframework/ibistesttool/IbisDebuggerAdvice.java

```diff
@@ -199,22 +199,16 @@ public PipeRunResult debugValidatorInputOutputAbort(ProceedingJoinPoint proceedi
 		return pipeRunResult;
 	}
 
-	private enum SenderReturnType {
-		MESSAGE,
-		PIPERUNRESULT,
-		SENDERRESULT;
-	}
-
-	private <M> M debugSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Message message, PipeLineSession session, int messageParamIndex, SenderReturnType returnType) throws Throwable {
+	private SenderResult debugSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Message message, PipeLineSession session, int messageParamIndex) throws Throwable {
 		if (!isEnabled()) {
-			return (M)proceedingJoinPoint.proceed();
+			return (SenderResult) proceedingJoinPoint.proceed();
 		}
-		ISender sender = (ISender)proceedingJoinPoint.getTarget();
+		ISender sender = (ISender) proceedingJoinPoint.getTarget();
 
 		String correlationId = getCorrelationId(session);
 		message = ibisDebugger.senderInput(sender, correlationId, message);
 
-		M result = null; // result can be PipeRunResult (for StreamingSenders) or Message (for all other Senders)
+		SenderResult senderResult;
```

Comment:
So, it is always a SenderResult currently?

Reply:
Yup!